uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1024.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+trait CRM_HRLeaveAndAbsences_Upgrader_Step_1024 {
+  
+  /**
+   * Update leave request templates to system workflow message
+   *
+   * @return bool
+   */
+  public function upgrade_1024() {
+    $templateName = 'msg_tpl_workflow_leave';
+    $result = $this->up1024_createLeaveWorkflowOptionGroup($templateName);
+    
+    if ($result['count'] > 0) {
+      $this->up1024_createLeaveWorkflowOptionValue($result['id']);
+    }
+    
+    return TRUE;
+  }
+  
+  /**
+   * Create Leave Message Template Workflow Option Group.
+   *
+   * @param $templateName
+   *
+   * @return array
+   */
+  public function up1024_createLeaveWorkflowOptionGroup($templateName) {
+    $workflow = 'Message Template Workflow for Leave';
+    
+    $result = civicrm_api3('OptionGroup', 'create', [
+      'name' => $templateName,
+      'title' => $workflow,
+      'description' => $workflow,
+    ]);
+    CRM_Core_PseudoConstant::flush();
```

Comment:
Please add a comment here explaining why this flush is necessary

Reply:
Alright. Will add comment.